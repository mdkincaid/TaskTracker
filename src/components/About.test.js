// react
import React from 'react';
import { Router, Route } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';

// third-party
import { createMemoryHistory } from 'history';

// project
import About from './About';

describe('About', () => {
	it('should render', () => {
		// arrange
		const history = createMemoryHistory();
		const route = '/';

		// act
		history.push(route);
		const { container } = render(
			<Router history={history}>
				<Route path='/about'>
					<About />
				</Route>
			</Router>
		);

		// assert
		expect(container).toMatchSnapshot();
	});

	it('should dispaly proper version of 1.2.0', () => {
		// arrange
		const history = createMemoryHistory();
		const route = '/';

		// act
		history.push(route);
		render(
			<Router history={history}>
				<Route>
					<About />
				</Route>
			</Router>
		);

		// assert
		expect(screen.getByText('Version 1.2.0')).toBeInTheDocument();
	});
});
