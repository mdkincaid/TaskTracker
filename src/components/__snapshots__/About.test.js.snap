// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should render 1`] = `<div />`;
